job.go

```diff
@@ -207,3 +207,18 @@ func (j *Job) String() string {
 		"{project=%s params=%s group=%s id=%s}",
 		j.Project, j.Params, j.Group, j.ID[:7])
 }
+
+// State determines the job's current state by using it's path in the filesystem.
+func (j *Job) State(path, project, id string) (string, error) {
+	pPath := filepath.Join(path, project, "pending", id)
+	rPath := filepath.Join(path, project, "ready", id)
+	_, err := os.Stat(pPath)
+	if err == nil {
```

Comment:
What if error is non nil but also not a "not exist" error?

Reply:
It will return the error.

Reply:
You're right, nvm.